server/app/mutations/grid_services/common.rb

```diff
@@ -352,6 +352,7 @@ def common_validations
               integer :initial_delay, default: 10
             end
           end
+          string :stop_signal
```

Comment:
Maybe also validate the signal here to catch mistakes earlier, rather than having to wait for the deploy fail - although looking at Docker's [`ParseSignal`](https://github.com/moby/moby/blob/a3efe9722f34af5cf4443fe3a5c4e4e3e0457b54/pkg/signal/signal.go#L31-L44), it's difficult to validate this meaningfully without encoding a complete list of all signals. I'm not sure if the regex for matching the supported values isn't just more trouble than it's worth:

```
string :stop_signal, matches: /\A((SIG)?([A-Z0-9]+|RTMIN\+\d+|RTMAX-\d+)|\d+)\z/i
```

Reply:
I thought about validating the input but could not (at least quickly) find conclusive documentation what signals are valid (over all possible OS's using signals).

I could try looking into docker-compose's implementation, whether they have any validation rules.

Reply:
> @SpComb : About the regex validation:
> At least looking through the SignalMaps Docker uses, that regex you showed could be a valid option to at least prevent totally incorrect signals. It won't be perfect, but it might be overkill to maintain lists similar to these ones Docker has: ...
> What do you think?

Yeah, definitely overkill to hardcode the list of valid signals. If the regex looks okay and doesn't end up accidentally excluding any valid signals, then it wouldn't hurt to include it... it will just also need `GridServices::Create` specs for it, not convinced if worth the effort. I'm ready to accept this without it, but great if you want to write those up.